pkg/apis/v1/nodepool_conversion_test.go

```diff
@@ -0,0 +1,536 @@
+	})
+
+	It("should convert v1 nodepool metadata", func() {
+		v1nodepool.ObjectMeta = test.ObjectMeta()
+		Expect(v1nodepool.ConvertFrom(ctx, v1beta1nodepool)).To(Succeed())
+		Expect(v1beta1nodepool.ObjectMeta).To(BeEquivalentTo(v1nodepool.ObjectMeta))
+	})
+	Context("NodePool Spec", func() {
+		It("should convert v1 nodepool weights", func() {
+			v1nodepool.Spec.Weight = lo.ToPtr(int32(62))
+			Expect(v1nodepool.ConvertTo(ctx, v1beta1nodepool)).To(Succeed())
+			Expect(lo.FromPtr(v1beta1nodepool.Spec.Weight)).To(Equal(int32(62)))
+		})
+		It("should convert v1 nodepool limits", func() {
+			v1nodepool.Spec.Limits = Limits{
+				v1.ResourceCPU:    resource.MustParse("5"),
+				v1.ResourceMemory: resource.MustParse("14145G"),
+			}
+			Expect(v1nodepool.ConvertTo(ctx, v1beta1nodepool)).To(Succeed())
+			for _, resource := range lo.Keys(v1nodepool.Spec.Limits) {
+				Expect(v1beta1nodepool.Spec.Limits[resource]).To(Equal(v1nodepool.Spec.Limits[resource]))
+			}
+		})
+		Context("NodeClaimTemplate", func() {
+			It("should convert v1 nodepool template labels", func() {
+				v1nodepool.Spec.Template.Labels = map[string]string{
+					"test-key-1": "test-value-1",
+					"test-key-2": "test-value-2",
+				}
+				Expect(v1nodepool.ConvertTo(ctx, v1beta1nodepool)).To(Succeed())
+				for key := range v1nodepool.Spec.Template.Labels {
+					Expect(v1beta1nodepool.Spec.Template.Labels[key]).To(Equal(v1nodepool.Spec.Template.Labels[key]))
+				}
+			})
+			It("should convert v1 nodepool template annotations", func() {
+				v1nodepool.Spec.Template.Annotations = map[string]string{
+					"test-key-1": "test-value-1",
+					"test-key-2": "test-value-2",
+				}
+				Expect(v1nodepool.ConvertTo(ctx, v1beta1nodepool)).To(Succeed())
+				for key := range v1nodepool.Spec.Template.Annotations {
+					Expect(v1beta1nodepool.Spec.Template.Annotations[key]).To(Equal(v1nodepool.Spec.Template.Annotations[key]))
+				}
+			})
+			It("should convert v1 nodepool template taints", func() {
+				v1nodepool.Spec.Template.Spec.Taints = []v1.Taint{
+					{
+						Key:    "test-key-1",
+						Value:  "test-value-1",
+						Effect: v1.TaintEffectNoExecute,
+					},
+					{
+						Key:    "test-key-2",
+						Value:  "test-value-2",
+						Effect: v1.TaintEffectNoSchedule,
+					},
+				}
+				Expect(v1nodepool.ConvertTo(ctx, v1beta1nodepool)).To(Succeed())
+				for i := range v1nodepool.Spec.Template.Spec.Taints {
+					Expect(v1beta1nodepool.Spec.Template.Spec.Taints[i].Key).To(Equal(v1nodepool.Spec.Template.Spec.Taints[i].Key))
+					Expect(v1beta1nodepool.Spec.Template.Spec.Taints[i].Value).To(Equal(v1nodepool.Spec.Template.Spec.Taints[i].Value))
+					Expect(v1beta1nodepool.Spec.Template.Spec.Taints[i].Effect).To(Equal(v1nodepool.Spec.Template.Spec.Taints[i].Effect))
+				}
+			})
+			It("should convert v1 nodepool template startup taints", func() {
+				v1nodepool.Spec.Template.Spec.StartupTaints = []v1.Taint{
+					{
+						Key:    "test-key--startup-1",
```

Comment:
nit doesn't really matter though
```suggestion
						Key:    "test-key-startup-1",
```